portfolio/src/main/webapp/script.js

```diff
@@ -40,6 +44,10 @@ async function showCommentForm() {
   } else {
     document.getElementById('comments-form').style.display = 'none';
   }
+  const moodForm = document.getElementById('select-mood');
+  for (let i = 0; i < moodForm.length; i++) {
+    moodForm[i].value = moods[i];
+  }
```

Comment:
Now, you fill in the values of the moods programmatically.
You assume here that the two arrays have the same length (the list in the html file and the array here in js).
You also assume that they are initialized with the same moods. Otherwise the user would pick mood X from the dropdown, but the underlying value is mood Y (which is then sent to the server).
Can't you initialize the whole dropdown here?